Sources/xcodeproj/Objects/Project/PBXObjectParser.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+final class PBXObjectParser {
+    
+    let reference: String
+    let userInfo: [CodingUserInfoKey: Any]
+    
+    init(reference: String, userInfo: [CodingUserInfoKey: Any]) {
+        self.reference = reference
+        self.userInfo = userInfo
+    }
+    
+    // swiftlint:disable function_body_length
+    public func parse(dictionary: [String: Any]) throws -> PBXObject {
+        let decoder = XcodeprojJSONDecoder()
+        decoder.userInfo = userInfo
+        var mutableDictionary = dictionary
+        mutableDictionary["reference"] = reference
+        let data = try JSONSerialization.data(withJSONObject: mutableDictionary, options: [])
```

Comment:
This is an unnecessary step that was introduced to leverage the decoders but that we can get rid of in a follow-up PR and get another performance improvement.

Reply:
In follow-up PR